crates/sound/src/lib.rs

```diff
@@ -223,6 +223,10 @@ impl SoundConfig {
     pub fn get_socket_path(&self) -> String {
         String::from(&self.socket)
     }
+
+    pub fn get_backend_name(&self) -> String {
+        String::from(&self.audio_backend_name)
+    }
```

Comment:
the audio_backend should probably replaced to take BackendType & returned here instead of a string

Reply:
`get_backend_name` is a getter for the `SoundConfig.audio_backend_name`, which is a `String` declared as private.
I've just changed the type for `SoundArgs.backend` member.
Also, it is only used in a single test at the moment.
So I think the method returning a `String` is ok as is.

Now, another discussion would be if we want the backend stored in `SoundConfig` to be `BackendType`. I think it may be nice, completely dropping the `str`/`String` type even for the patterns here: https://github.com/virtio-sound/vhost-device/blob/virtio-sound/crates/sound/src/audio_backends.rs#L53
And matching `BackendType`s instead.

Reply:
Right, that is why I asked this. If an enum is introduced to define the backend to use, it should be used elsewhere as well.

Reply:
If I take this road, though, I would have to move the `BackendType` declaration to the `lib` module, including the clap `ValueEnum` attribute, which is a bit annoying as it is a CLI option that belongs to the binary.

Because otherwise, I cannot import stuff declared in main to other modules.

So I wonder if we really want to take this path.

Reply:
I cannot implement an imported trait on an imported struct. But ok, I'll use the `ValueEnum` attribute in `lib`. Let's see how it looks.

Reply:
ooh indeed. Actually i think that is fine as clap is not explicitly set as a binary only dependency :)

Reply:
True :)

Done with the last push!